DM-4351

jonathansick opened the pull request.

jonathansick committed:
Create document outline

For DM-4351.

jonathansick committed:
Write section describing niche of technotes

For DM-4351.

jonathansick committed:
Describe design choices for technotes

For DM-4351.

jonathansick committed:
Ignore _build/ directory

jonathansick committed:
Describe lsst-technote-bootstrap impl

For DM-4351.

jonathansick committed:
Add a roadmap for technote platform.

For DM-4351.

jonathansick committed:
Describe metadata and propose extensions

For DM-4351.

jonathansick committed:
Edits of first draft

For DM-4351.

jhoblitt commented on index.rst:
```diff
@@ -1,40 +1,378 @@
+We describe the design and architecture of the DM subsystem in change-controlled design documents.
+We publish articles in journals and conference proceedings where appropriate.
+Finally, we have several Confluence wikis that traditionally served as a catch-all for DM documentation.
+Despite this array of platforms, a use case was still unserved and this technote platform was conceived to meet that need.
```
Should this be a bullet-ed list?

jhoblitt commented:
This looks good after a quick skim.

How about enabling travis?

jonathansick committed:
Add travis checks for HTML builds

For DM-4351.

jonathansick committed:
Use wikipedia for DOI background link

timj commented on index.rst:
```diff
@@ -1,40 +1,378 @@
-  See http://docs.lsst.codes/en/latest/development/docs/rst_styleguide.html
-  for a guide to reStructuredText writing.
+Technical notes (hereafter, *technotes*) are a new documentation medium for `LSST Data Management <http://dm.lsst.org>`_ staff to publish stand-alone documents that are native to the web.
+Technotes are written in reStructuredText, version controlled on GitHub, built with the same stack as DM's software documentation, and made universally citeable with `digital object identifiers (DOIs) <https://en.wikipedia.org/wiki/Digital_object_identifier>`_ provided through Zenodo_.
```
Can `reStructuredText` link to a page on how to use it and what it is? Which I see you do do later on.

timj commented on index.rst:
```diff
@@ -1,40 +1,378 @@
-  Subsubsections
-  ^^^^^^^^^^^^^^
+In SQuaRE, the Science Quality and Reliability Engineering group, we realized that we often wanted to write stand-alone documents to describe our work.
+These documents didn't fit the conversational nature of `Community forum`_ topics, nor did they qualify for DM's change-controlled design document series, or fit into academic journals or the arXiv_.
```
How come "Community forum" has back tick quotes around it yet "arXiv" does not? They both seem to translate to external URLs using the same text style.

jonathansick replied:
The important part is the trailing underscore to make a link. Back ticks on a single word link end up being redundant.

timj commented on index.rst:
```diff
@@ -1,40 +1,378 @@
+Technotes were thus built on the same technology stack as the reStructuredText-based design documents, but re-purposed for general use by DM team members.\ [#]_ 
 
-     Caption text.
+.. [#] In fact, this technote applies equally to describing the new reStructuredText-based design document platform as it does the technote platform. The only meaningful difference is that design documents must be approved by a control board, and are ultimately deposited in LSST's Docushare.
```
link to Docushare.

timj commented on index.rst:
```diff
@@ -1,40 +1,378 @@
+`Markdown <http://daringfireball.net/projects/markdown/>`_ is a hugely popular format among developers, in part because of its intuitive simplicity.
+However, this simplicity has prompted developers to `add non-standard extensions to the format <http://commonmark.org>`_ to their markup to add semantics and style to content.
+
+Instead, reStructuredText is an ideal markup format for technotes.
```
Why only make this clickable once?

timj commented on index.rst:
```diff
@@ -1,40 +1,378 @@
 
-  See http://docs.lsst.codes/en/latest/development/docs/rst_styleguide.html
-  for a guide to reStructuredText writing.
+Technical notes (hereafter, *technotes*) are a new documentation medium for `LSST Data Management <http://dm.lsst.org>`_ staff to publish stand-alone documents that are native to the web.
```
Maybe say "(DM)" after "Data Management" here as "DM" is used later (yes, I know everyone who cares will know already).

timj commented on index.rst:
```diff
@@ -1,40 +1,378 @@
 
-  Sections
-  ========
+We currently use two technote series. The 'DMTN' is for general use within Data Management.
```
"Data Management Tech Note" (or use `:abbr:` so it has a mouse over).

timj commented on index.rst:
```diff
@@ -1,40 +1,378 @@
+Technotes are native to the web
+-------------------------------
+
+Although a preponderance of astronomy literature is published as PDFs, and many management documents as Word documents, those formats are archaic solutions from an era when documents were read in print.
```
Drop "archaic". It's inflammatory and Word and PDF are perfectly good solutions for what they were designed for. How about: "those formats are optimized for print media"?

timj commented on index.rst:
```diff
@@ -1,40 +1,378 @@
+-------------------------------
+
+Although a preponderance of astronomy literature is published as PDFs, and many management documents as Word documents, those formats are archaic solutions from an era when documents were read in print.
+Modern HTML, CSS and JavaScript allow for better, more innovative, and frictionless reading experiences than PDFs and Word documents can provide.
```
This sentence would not pass muster on Wikipedia :smile:

jmatt commented:
@timj Have you considered becoming a copy editor on the side? At the speed you went through this document you could probably hold down five editor jobs.

timj commented on index.rst:
```diff
@@ -1,40 +1,378 @@
+
+Instead, reStructuredText is an ideal markup format for technotes.
+reStructuredText's syntax and build process was designed to be extended, and in fact, those extensions are made in DM's primary language, Python.
+Because the Python community has adopted reStructuredText as its official markup language, a our technote platform can take advantage of excellent open source tools, such as the Sphinx build system and the Read the Docs documentation hosting site.
```
drive home the point by mentioning Astropy here, so familiar to a new generation of astronomers.

timj commented on index.rst:
```diff
@@ -1,40 +1,378 @@
+Technotes are single-page documents
+-----------------------------------
+
+This limitation is made purposefully to limit the scope of technotes, make it possible to easily print a technote in its entirely, and allow readers to use their browser's search feature to navigate a document.
```
Although, ironically, if you are talking about printing you had better be able to format it in PDF properly. Nothing worse than printing an HTML page and seeing it split a line in half horizontally across pages.

timj commented on index.rst:
```diff
@@ -1,40 +1,378 @@
+Since they are standalone, self-encapsulated documents, technotes can be easily archived and assigned digital object identifiers, DOIs.
+DOIs allow technotes to be easily cited in scientific literature.
+The benefit of a citeable-trail of DM documentation is that LSST publications can more accurately describe DM's engineering work.
+We use Zenodo_ as an archive and DOI provider, taking advantage of its GitHub integration.
```
There should be a tech note on how to do that, and to ensure that the viewable HTML/PDF becomes easily visible and not embedded inside a tar ball of the source if the DOI is followed.

jonathansick replied:
That sounds more like software user documentation (this is already in the lsst-technote-boilerplate README), and will eventually be automated completely. The automation will need to solve the viewing problem with DOI links, that's true.

timj replied:
I had not realised until later on in the document that you refer to instructions.

timj commented on index.rst:
```diff
@@ -1,40 +1,378 @@
+Technotes are versioned
+-----------------------
+
+Unlike scientific publications that are published once, technotes can and should be updated when appropriate.
```
The "can and should be updated when appropriate" phrasing makes it hard to say whether you approve of tech notes that provide information and then never change. Also may want to mention in the DOI tech note how DOIs deal with versioned documents.

timj commented on index.rst:
I'm not sure this sentence is needed. It may well be out of date in a week as the bootstrap is developed. It can stay if this is a deliberate hedge to warn people that the bootstrap is in beta.

timj commented on index.rst:
Since these are files and functions maybe use `:file:` and the like.

``` rst
    :file:`conf.py`
```

jonathansick commented on index.rst:
Given that I need to work on the Stack doc platform, technote platform development will probably get backburnered for the next few months.

timj commented on index.rst:
ok, now I see the instructions.

timj commented on index.rst:
Guaranteed to forget that.

timj commented:
Looks great to me.

jonathansick commented:
Thanks @timj! I'll put these fixes in.

jonathansick committed:
Addressed most of timj's comments

- better linking to software
- use file and abbr roles when appropriate
- be kinder to PDF allies
- updated text on versioning

For DM-4351.

jonathansick committed:
Update proposed metadata

- Add deprecated field, and changes field
- Note that version and last_revised *could* be machine-generated in the
  future

For DM-4351.

jonathansick committed:
Add Acknowledgements section

jonathansick committed:
Use documenteer==0.1.0

jonathansick committed:
Pre-v1.0 proofreading pass

For DM-4351.

jonathansick committed:
Mark v1.0

For DM-4351.
